libraries/mbed/targets/cmsis/TARGET_NORDIC/TARGET_MCU_NRF51822/system_nrf51.c

```diff
@@ -61,6 +61,20 @@ void SystemCoreClockUpdate(void)
 
 void SystemInit(void)
 {
+#if defined(TARGET_NRF_32MHZ_XTAL)
```

Comment:
Can you provide brief description what it does?

I don't understand these 2 lines : ```if (*(uint32_t *)0x10001008 == 0xFFFFFFFF)``` and ```NVIC_SystemReset();```